packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -993,6 +993,102 @@ class DefaultTextStyleTransition extends AnimatedWidget {
 /// ** See code in examples/api/lib/widgets/transitions/animated_builder.0.dart **
 /// {@end-tool}
 ///
+/// {@template flutter.flutter.animatedbuilder_changenotifier.rebuild}
+/// ## Granular rebuilds with AnimatedBuilder and ChangeNotifier
+///
+/// A [State.setState] method call requires Flutter to entirely rebuild the
+/// current widget plus a variable portion of the subtree, depending on the
+/// depth and the structure.
+///
+/// To avoid unnecessary work, the [AnimatedBuilder] class can be used along
+/// with a [ChangeNotifier] to only rebuild certain portions a widget,
+/// leaving other parts and the subtree untouched.
+///
+/// Since a [ChangeNotifier] is a subtype of [Listenable], it can be used in
+/// an [AnimatedBuilder] to listen for changes. For example, to keep track
+/// of the state of a simple counter, a [ChangeNotifier] can be created and
+/// exposed to the widgets below using an [InheritedWidget].
+///
+/// {@tool snippet}
```

Comment:
These should probably use the `sample` or `dartpad` notation. That will subject them to analysis and testing, and the later will make them interactive examples in the docs.

Reply:
I've just pushed a change where I removed `{@tool snippet}` because I saw it caused problems to the linux CI step